node/modules/blockstore.go

```diff
@@ -31,17 +36,77 @@ func UniversalBlockstore(lc fx.Lifecycle, mctx helpers.MetricsCtx, r repo.Locked
 	return bs, err
 }
 
-// StateBlockstore is a hook to overlay caches for state objects, or in the
-// future, to segregate the universal blockstore into different physical state
-// and chain stores.
-func StateBlockstore(lc fx.Lifecycle, mctx helpers.MetricsCtx, bs dtypes.UniversalBlockstore) (dtypes.StateBlockstore, error) {
+func BadgerHotBlockstore(lc fx.Lifecycle, r repo.LockedRepo) (dtypes.HotBlockstore, error) {
+	path, err := r.SplitstorePath()
+	if err != nil {
+		return nil, err
+	}
+
+	path = filepath.Join(path, "hot.badger")
```

Comment:
I want to keep the hotstore type in the name so that we don't get ridiculous bugs with distinguishing it.
We currently only use badger, but the experimental lmdb branch can change this.

Reply:
Yes, we will likely change the hot store library at some point (go-storethehash, lmdb, go-nudb, etc.)

Reply:
Important detail is that it need to be able to delete, so storethehash and nudb are not viable.